corehq/motech/migrations/0017_connectionsettings_use_aes_cbc_encryption.py

```diff
@@ -0,0 +1,121 @@
+        client_secret='',
+        last_token_aes=''
+    )
+
+    updated_connections = []
+    for connection in connect_settings_to_update:
+        connection.password = _reencrypt_or_encrypt_value_with_cbc(connection.password)
+        connection.client_secret = _reencrypt_or_encrypt_value_with_cbc(connection.client_secret)
+        connection.last_token_aes = _reencrypt_value_with_cbc(connection.last_token_aes)
+        updated_connections.append(connection)
+
+    ConnectionSettings.objects.bulk_update(updated_connections, ['password', 'client_secret', 'last_token_aes'])
+
+
+def _reencrypt_or_encrypt_value_with_cbc(value):
+    if value == '':
+        return ''
+    if value.startswith(f'${ALGO_AES_CBC}$'):
+        return value
+
+    if value.startswith(f'${ALGO_AES}$'):
+        try:
+            return reencrypt_ecb_to_cbc_mode(value, f'${ALGO_AES}$')
+        except AesEcbDecryptionError:
+            return ''
+    else:
+        ciphertext = b64_aes_cbc_encrypt(value)
+        return f'${ALGO_AES_CBC}${ciphertext}'
+
+
+def _reencrypt_value_with_cbc(value):
+    if value == '':
+        return ''
+    if value.startswith(f'${ALGO_AES_CBC}$'):
+        return value
+
+    if value.startswith(f'${ALGO_AES}$'):
+        prefix = f'${ALGO_AES}$'
+    else:
+        prefix = None
+    try:
+        return reencrypt_ecb_to_cbc_mode(value, prefix)
+    except AesEcbDecryptionError:
+        return ''
+
+
+def revert_api_settings(apps, schema_editor):
+    ConnectionSettings = apps.get_model("motech", "ConnectionSettings")
+
+    connect_settings_to_revert = ConnectionSettings.objects.exclude(
+        password__startswith=f'${ALGO_AES}$',
+        client_secret__startswith=f'${ALGO_AES}$',
+        last_token_aes__startswith=f'${ALGO_AES}$',
+    ).exclude(
+        password='',
+        client_secret='',
+        last_token_aes=''
+    )
+
+    updated_connections = []
+    for connection in connect_settings_to_revert:
+        connection.password = _revert_reencrypted_value_to_ecb(connection.password)
+        connection.client_secret = _revert_reencrypted_value_to_ecb(connection.client_secret)
+        connection.last_token_aes = _revert_last_token_to_ecb(connection.last_token_aes)
+        updated_connections.append(connection)
+
+    ConnectionSettings.objects.bulk_update(updated_connections, ['password', 'client_secret', 'last_token_aes'])
+
+
+def _revert_reencrypted_value_to_ecb(value):
+    if value == '':
+        return ''
+    elif value.startswith(f'${ALGO_AES_CBC}$'):
+        return reencrypt_cbc_to_ecb_mode(value, f'${ALGO_AES_CBC}$')
+    else:
+        return value
+
+
+def _revert_last_token_to_ecb(value):
+    if value == '':
+        return ''
+    elif value.startswith(f'${ALGO_AES_CBC}$'):
+        encrypted_val = reencrypt_cbc_to_ecb_mode(value, f'${ALGO_AES_CBC}$')
+        # last_token_aes is originally only stored without a prefix
+        return encrypted_val.split('$', 2)[2]
+    else:
+        return value
```

Comment:
What do you think of combining this two function, with an additional parameter `without_ecb_prefix` ?

Reply:
Thanks, I like that [add9c8b](https://github.com/dimagi/commcare-hq/pull/35641/commits/add9c8b000d66e7bcbd13b4282c2b22878a0fb76)